ksqldb-engine/src/test/java/io/confluent/ksql/planner/plan/PullFilterNodeTest.java

```diff
@@ -971,13 +1025,29 @@ public void shouldThrowIfNonEqualComparison() {
             expression,
             metaStore,
             ksqlConfig,
-            false
+            false,
+            plannerOptions
         ));
 
     // Then:
     assertThat(e.getMessage(), containsString("Bound on key columns '[`K` INTEGER KEY]' must currently be '='."));
   }
 
+  @SuppressWarnings("unchecked")
+  @Test
+  public void shouldExtractConstraintWithLiteralRange_tableScan() {
```

Comment:
when we support key range queries this should no longer require table scan, right?